sql/catalyst/src/main/scala/org/apache/spark/sql/types/StructType.scala

```diff
@@ -407,8 +407,12 @@ case class StructType(fields: Array[StructField]) extends DataType with Seq[Stru
 
   override def catalogString: String = {
     // in catalogString, we should not truncate
-    val fieldTypes = fields.map(field => s"${field.name}:${field.dataType.catalogString}")
-    s"struct<${fieldTypes.mkString(",")}>"
+    val stringConcat = new StringUtils.StringConcat()
+    fields.init.foreach { field =>
+      stringConcat.append(s"${field.name}:${field.dataType.catalogString}, ")
```

Comment:
The tests failed because of the space in `catalogString}, "`

Reply:
Thanks. :)